File: app/src/main/java/com/game/tama/command/CommandWait.java
package com.game.tama.command;

import android.util.Log;

import com.game.engine.Time;
import com.game.tama.core.thing.pet.Pet;
import com.game.tama.core.world.World;

public class CommandWait extends Command
{

    private final float waitTimeMs;
    private float timeMs = 0f;

    public CommandWait(final float waitTimeMs)
    {
        this.waitTimeMs = waitTimeMs;
    }

    @Override
    public void start(final Pet pet, final World world)
    {
        super.start(pet, world);
        Log.d(getClass().getCanonicalName(), "pet waiting $waitTimeMs");
        update.update(pet, world);
    }

    @Override
    public void doing(final Pet pet, final World world)
    {
        timeMs += Time.deltaTimeMs();
        if (timeMs >= waitTimeMs)
        {
            complete();
        }
    }
}

File: app/src/main/java/com/game/tama/command/CommandWait.kt
package com.game.tama.command

import android.util.Log
import com.game.engine.DisplayAdapter
import com.game.engine.GameLoop
import com.game.tama.thing.pet.Pet
import com.game.tama.core.World

public class CommandWait(private val waitTimeMs: Float) : Command()
{

    private var timeMs: Float = 0f;

    public override fun start(pet: Pet, world: World)
    {
        super.start(pet, world);
        Log.d(javaClass.canonicalName, "pet waiting $waitTimeMs");
        update(pet, world);
    }

    public override fun doing(pet: Pet, world: World)
    {
        timeMs += GameLoop.deltaTimeMs;
        if (timeMs >= waitTimeMs)
        {
            complete();
        }
    }

    override fun hardCancel()
    {
        super.hardCancel();
    }

    override fun draw(d: DisplayAdapter)
    {

    }
}